week-2-practise/level-3/5-indexOf.js

```diff
@@ -4,9 +4,21 @@
 // - insert `newValue` at that index
 // - NOT change the original array
 // - return a new array with the replacement value inserted
+function firstPartOfArr(arr, index) {
+  return arr.slice(0, index);
+}
+
+function lastPartOfArr(arr, index) {
+  return arr.slice(index + 1, arr.length);
+}
 
 function replace(arr, valueToReplace, newValue) {
-  return; // complete this statement
+  let replacedArr = [];
+  let removedIndex = arr.indexOf(valueToReplace);
+  let replacedArrFirst = firstPartOfArr(arr, removedIndex);
+  replacedArr[removedIndex] = newValue;
+  let replacedArrLast = lastPartOfArr(arr, removedIndex);
+  return replacedArrFirst.concat(replacedArr[removedIndex], replacedArrLast);
```

Comment:
I think this exercise is quite challenging, so well done on completing it!

Reply:
Thanks for your encouraging words Ali.